aggregator/handler/writer/sharded.go

```diff
@@ -156,16 +164,36 @@ func (w *shardedWriter) encode(
 	bufferedEncoder := encoder.Encoder()
 	buffer := bufferedEncoder.Buffer()
 	sizeBefore := buffer.Len()
-	if err := encoder.EncodeChunkedMetricWithStoragePolicy(mp); err != nil {
+
+	// Encode data.
+	var (
+		includeEncodingTime bool
+		err                 error
+	)
+	if w.includeEncodingTime {
```

Comment:
nit: would it perhaps be preferable to use `rand` to determine if we should include the encoding time? My concern is that if we sample every N metrics and there's some kind of bias in the latency of metrics that is dependent on the order of the metrics we may feel to detect it.

Reply:
Sure can do.